internal/provider/provider.go

```diff
@@ -69,9 +77,14 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
 	ServerURL := data.ServerURL.ValueString()
 	ClientID := data.ClientID.ValueString()
 	ClientSecret := data.ClientSecret.ValueString()
+	TenantDomain := data.TenantDomain.ValueString()
 
 	if ServerURL == "" {
-		ServerURL = "https://{tenantDomain}.conductor.one"
+		if TenantDomain == "" {
+			url := strings.Split(ClientID, "@")[1]
+			TenantDomain = strings.Split(url, ".")[0]
```

Comment:
we should use the SDK function for this logic?  doesn't it fully handle splitting the Client ID?